src/api/routers/topic.ts

```diff
@@ -22,14 +22,23 @@ export const topicRouter = router({
     )
     .query(async (opts) => {
       const isCreator = opts.input.username === opts.ctx.user?.username;
+      const normalizedTitle = normalizeTitle(decodeURIComponent(opts.input.title));
 
-      return await xprisma.topic.findFirst({
-        where: {
-          title: opts.input.title,
-          creatorName: opts.input.username,
-          visibility: isCreator ? undefined : { not: "private" },
-        },
-      });
+      const topics = await xprisma.$queryRaw<
+        { id: number; title: string; creatorName: string; visibility: string }[]
+      >`
+        SELECT id, title, "creatorName", visibility
```

Comment:
```suggestion
        Topic
      >`
        SELECT *
```
In the shared method I mention above, I think it's fine to return all topic columns - then we can also use prisma's generated `Topic` type.